File: ayase-server/tools/2048.ts
/**
 * https://2048game.com/
 */
import { Danmaku } from "~~/types"

// 记录上机用户
let currentUser: string | null = null

export default function game2048(danmaku: Danmaku) {
  if (danmaku.text === '2048上机') {
    currentUser = danmaku.uname
    console.log(`2048用户: [${danmaku.uname}] 已上机！`)
  }

  if (danmaku.uname === '诡锋Vincent' && danmaku.text === '强制下机') {
    currentUser = null
    console.log(`2048 -- 管理员已强制用户下机！`)
  }

  if (!currentUser) {
    return
  }

  if (currentUser === danmaku.uname) {
    if (danmaku.text === '2048下机') {
      currentUser = null
      console.log(`2048用户: [${danmaku.uname}] 已下机！`)
    }
    else {
      console.log(`2048用户: [${danmaku.uname}] 发送指令`, danmaku)
      const { text } = danmaku
      // switch (text) {
      //   case '上':
      //     robot.keyTap('up')
      //     break
      //   case '下':
      //     robot.keyTap('down')
      //     break
      //   case '左':
      //     robot.keyTap('left')
      //     break
      //   case '右':
      //     robot.keyTap('right')
      //     break
      //   default:
      //     break
      // }
    }
  }
}

File: ayase-handler/2048.py
import asyncio
from websockets.asyncio.client import connect
import json
import pyautogui

""" https://2048game.com/ """

async def game_2048():
    uri = "ws://localhost:8081/websocket"
    
    try:
        async with connect(uri) as websocket:
            print(f"Websocket connected: {uri}")
            
            while True:
                recv_msg = await websocket.recv()
                recv_json = json.loads(recv_msg)
                print(f"Received: {recv_json}")

                # 检查是否有退出消息
                if recv_json.get("type") == "exit":
                    print("收到退出消息，关闭连接")
                    break
                    
                if recv_json.get("type") == "2048":
                    text = recv_json.get("text", "")
                    if text == "上":
                        pyautogui.press("up")
                    elif text == "下":
                        pyautogui.press("down")
                    elif text == "左":
                        pyautogui.press("left")
                    elif text == "右":
                        pyautogui.press("right")
                        
    except Exception as e:
        print(f"WebSocket错误: {e}")

if __name__ == "__main__":
    asyncio.run(game_2048())